File: coreServer/src/main/java/com/example/coreServer/dto/SendMessageRequest.java
package com.example.coreServer.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SendMessageRequest {
    private Long chatId;
    private Long senderId;
    private Long receiverId;
    private String text;
    private String replyToMessageId;
    private List<AttachmentDto> attachments;
}

File: CS Middleware/Shared/DTOs/MessageActionDTOs/SendMessageRequest.cs
using System;
using DTOs.ChatDTOs;

namespace DTOs.MessageActionDTOs;

public class SendMessageRequest {
    public required long ChatId { get; set; }
    public required long SenderId { get; set; }
    public long ReceiverId { get; set; }
    public String TextContent { get; set; } = String.Empty;
    public String ReplyToMessageId { get; set; } = String.Empty;
    public List<AttachmentDTO> Attachments { get; set; } = [];
}
